### PR TITLE
ACS-8123: add e2e tests for questions endpoint

### PR DESCRIPTION
**Pull request by Pawel-608**

### CODE REVIEW

File: e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/AskQuestionE2eTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+import org.wiremock.integrations.testcontainers.WireMockContainer;
+
+import org.alfresco.hxi_connector.common.test.docker.repository.AlfrescoRepositoryContainer;
+import org.alfresco.hxi_connector.common.test.docker.util.DockerContainers;
+
+@Testcontainers
+@SuppressWarnings("PMD.FieldNamingConventions")
+public class AskQuestionE2eTest
+{
+    static final Network network = Network.newNetwork();
```
Comment (tpage-alfresco) [inline e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/AskQuestionE2eTest.java:52]: This all looks good, but can't we reuse any of the E2E code we already have for the other components?

**Review by tpage-alfresco — APPROVED** [review]
I'm not sure the integration test is adding much value here. It's complex to set up and difficult to understand exactly what is being testing (vs what is being mocked). I like the E2E test though.

File: hxinsight-extension/src/integration-test/resources/rest/api/question-entity-resource-test-context.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  ~ You should have received a copy of the GNU Lesser General Public License
+  ~ along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+  ~ #L%
+  -->
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans
+          http://www.springframework.org/schema/beans/spring-beans-3.0.xsd">
+
+    <!--    Mocks    -->
```
[the remark this answers is not in the document]
Comment (Pawel-608) [reply hxinsight-extension/src/integration-test/resources/rest/api/question-entity-resource-test-context.xml]: yep, you're right - it's not used anymore - I deleted it

File: hxinsight-extension/src/integration-test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionsEntityResourceIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                "questionId": "%s"
+            }
+            """.formatted(QUESTION_ID);
+
+    @Container
+    @SuppressWarnings("PMD.FieldNamingConventions")
+    static final WireMockContainer hxInsightMock = DockerContainers.createWireMockContainer();
+
+    @MockBean
+    private HxInsightClientConfig hxInsightClientConfig;
```
[the remark this answers is not in the document]
Comment (Pawel-608) [reply hxinsight-extension/src/integration-test/java/org/alfresco/hxi_connector/hxi_extension/rest/api/QuestionsEntityResourceIntegrationTest.java]: As we spoke - I'll remove integration test for now and then we'll discuss what to do with it

File: e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/AskQuestionE2eTest.java (hunk 1 above)
Comment (Pawel-608) [reply e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/AskQuestionE2eTest.java:52]: idk, maybe we can but I would wait with that till we'll see repetitions to not waste time for something we won't use

**Pull request merged**